kalite/coachreports/api_views.py

```diff
@@ -154,9 +154,9 @@ def aggregate_learner_logs(request):
     output_dict["learner_events"] = [{
         "learner": log.user.get_name(),
         "complete": log.complete,
-        "struggling": log.struggling,
-        "progress": log.streak_progress,
-        "exercise": get_exercise_cache().get(log.exercise_id),
+        "struggling": getattr(log, "struggling", None),
+        "progress": getattr(log, "streak_progress", getattr(log, "progress", None)),
+        "content": get_exercise_cache().get(getattr(log, "exercise_id"), get_content_cache().get(getattr(log, "video_id", getattr(log, "content_id")), {})),
```

Comment:
Yes, that is indeed my intent.

If it were not being built inside a list comprehension, I would try to be a little less succinct. Also happy to comment it a bit if that's also helpful.